src/rabbit_auth_backend_oauth2.erl

```diff
@@ -77,28 +80,50 @@ user_login_authorization(Username, AuthProps) ->
 
 check_vhost_access(#auth_user{impl = DecodedToken},
                    VHost, _AuthzData) ->
+    TokenScopes = get_scopes(DecodedToken),
+    UserScopes = case check_picklock(TokenScopes) of
+        true ->
+            get_picklock_scopes(DecodedToken);
+        false -> 
+            TokenScopes
+    end,
```

Comment:
This sequence of code is repeated in several places, it should be in a dedicated function.

Reply:
cleaned up!